### PR TITLE
Minor: Improve docs for arrow-ipc, remove clippy ignore

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

N/A

# Rationale for this change
A deprecation warning says to use `IpcDataGenerator` but then there are no docstrings for that generator

# What changes are included in this PR?

1. Add a doc example for `IpcDataGenerator` 
2. Update the module docs with a link
3. Remove an uneeded clippy lint

# Are there any user-facing changes?

Better docs

### CODE REVIEW

File: arrow-ipc/src/lib.rs [hunk 1]
```diff
@@ -15,10 +15,7 @@
 // specific language governing permissions and limitations
 // under the License.
 
-//! Support for the Arrow IPC format
-
-// TODO: (vcq): Protobuf codegen is not generating Debug impls.
```
Comment (alamb) [inline arrow-ipc/src/lib.rs:20]: Drive by cleanup

File: arrow-ipc/src/writer.rs [hunk 2]
```diff
@@ -137,6 +137,34 @@ impl Default for IpcWriteOptions {
...
+/// # use arrow_ipc::writer::{DictionaryTracker, IpcDataGenerator, IpcWriteOptions};
+///
+/// // Create a record batch
+/// let batch = RecordBatch::try_from_iter(vec![
+///  ("col2", Arc::new(UInt64Array::from_iter([10, 23, 33])) as _)
+/// ]).unwrap();
+///
+/// // Error of dictionary ids are replaced.
+/// let error_on_replacement = true;
+/// let options = IpcWriteOptions::default();
```
Comment (alamb) [inline arrow-ipc/src/writer.rs:158]: This interface is somewhat unfortunate (like there is no state on IpcDataGenerator and the state is passed in via DictionaryTracker). 

While implementing #3389 I hope to provide something that handles all the encoding state so this interface will remain low level

**Review by alamb — COMMENTED** [review]
cc @Dandandan  as I believe you have an interest in this

File: arrow-ipc/src/writer.rs (hunk 2 above)
Comment (tustvold) [reply arrow-ipc/src/writer.rs:158]: I wonder if the state was at one point on IpcDataGenerator and it got factored out for some reason :thinking:

**Review by tustvold — APPROVED** [review]

**Review by viirya — APPROVED** [review]

File: arrow-ipc/src/lib.rs [hunk 3]
```diff
@@ -15,10 +15,7 @@
...
-#![allow(missing_debug_implementations)]
+//! Support for the [Arrow IPC Format](https://arrow.apache.org/docs/format/IPC.html)
```
Comment (viirya) [inline arrow-ipc/src/lib.rs:18]: Actually this page was moved. Maybe we can point to https://arrow.apache.org/docs/format/Columnar.html#format-columnar, or https://arrow.apache.org/docs/format/Columnar.html#serialization-and-interprocess-communication-ipc specifically?

File: arrow-ipc/src/writer.rs [hunk 4]
```diff
@@ -137,6 +137,34 @@ impl Default for IpcWriteOptions {
 }
 
 #[derive(Debug, Default)]
+/// Handles low level details of encoding [`Array`] and [`Schema`] into the
+/// [Arrow IPC Format](https://arrow.apache.org/docs/format/IPC.html)
```
Comment (viirya) [inline arrow-ipc/src/writer.rs:141]: ditto

File: arrow-ipc/src/writer.rs (hunk 2 above)
Comment (alamb) [reply arrow-ipc/src/writer.rs:158]: 🤷

**Pull request merged**